app/helpers/wowza_helper.rb

```diff
@@ -0,0 +1,44 @@
+module WowzaHelper
+  #---
+  # Builds Wowza URL
+  #
+  # @param obj_id
+  # @param file_set_id
+  # @param fedora_file
+  # @media_type: 'audio', 'video'
+  # @return string or nil
+  #---
+
+  def grabWowzaURL(obj_id, file_set_id, fedora_file, media_type, request_ip)
+    encrypted = encrypt_stream_name( obj_id, file_set_id, fedora_file, media_type, request_ip )
+    return Rails.configuration.wowza_baseurl + encrypted
+  end
+
+
+  ## video stream name encryption
+  def encrypt_stream_name( obj_id, file_set_id, fedora_file, media_type, ip )
+
+    # random nonce
+    nonce=rand(36**16).to_s(36)
+    while nonce.length < 16 do
+      nonce += "x"
+    end
+
+    # load key from environment variable
+    key = ENV.fetch('APPS_DHH_STREAMING_KEY') {'xxxxxxxxxxxxxxxx'}
```

Comment:
What do we want to happen if the streaming key doesn't exist? I suspect we don't want to return this bogus `'xx...'` value. We could not have a fallback, in which case I believe fetch with return a `KeyError`. That seems reasonable to me, since the key should exist. But curious what others think.